### PR TITLE
feat: migrate build tool from vue-cli 4 to vite(for lib build only) & export style modules

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skipping review is not a problem (```style```, ```chore``` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
  - [x] These changes require a usage change
- [x] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] Updated Storybook documents
- [x] Tested with console

### Description

- Migrate build tool from vue cli 4 to vite. No migration on storybook build.
  - change some common js files' extension from `.js` to `.cjs` (`.eslintrc.cjs`, `postcss.cjs`, `tailwind.config.cjs`, etc.)
  - change some .vue files to use `import()` rather then `require()` (`PMarkdown.vue`, `PTextEditor.vue`)
  - update `postcss.config` to use array for plugin, not a function(for vite)
  - add `vue-tsc` script for vue file type declaration (add to workflow also)
- export style modules (`styleVariables`, `colors`, `fonts`, `screens`)

### Things to Talk About

- must add `import '@spaceone/design-system/dist/style.css';` on main to apply styles of components.
- now, we can use npm link

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -12,9 +12,7 @@
 
 ### Checklist
 - [ ] Updated Storybook documents
-- <del>[ ] Tested with console</del> 
-- Currently, `npm link` does not work between console and mirinae
-- **Please check on console with released mirinae**
+- [ ] Tested with console(if usages are changed)
```
Comment (minseolee) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: 👍

File: .storybook/postcss.config.js [hunk 2]
```diff
@@ -1,25 +1,22 @@
...
 module.exports = {
     parser: 'postcss-comment',
-    plugins: () => {
-        const res = [
+    plugins:  [
             require('postcss-easy-import')({
                 path: ['../src', 'node_modules'],
             }),
-            require('tailwindcss'),
+            require('tailwindcss')({ config: path.resolve(__dirname, 'tailwind.config.js') }),
```
Comment (minseolee) [inline .storybook/postcss.config.js:9]: Is it ok using `.js` here?

File: build/css.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ const fs = require('fs');
 
 const postcss = require('postcss');
 
-const postcssConfig = require('../postcss.config');
+const postcssConfig = require('../postcss.config.cjs');
```
Comment (minseolee) [inline build/css.js:6]: It looks like postcss.config is using `.js` 🤔

File: package.json [hunk 4]
```diff
@@ -187,8 +190,12 @@
     "ts-loader": "^7.0.4",
     "tsc-alias": "^1.3.7",
     "typescript": "^4.6.3",
+    "vite": "^3.2.4",
+    "vite-plugin-stylelint": "^3.0.9",
+    "vite-plugin-vue2": "^2.0.2",
```
Comment (minseolee) [inline package.json]: `@vitejs/plugin-vue2` already exists, so I think you can remove this duplicated plugin, `vite-plugin-vue2`

**Review by minseolee — COMMENTED** [review]
Thanks a lot for your efforts!!! 
I left some comments

File: .storybook/postcss.config.js (hunk 2 above)
Comment (WANZARGEN) [reply .storybook/postcss.config.js:9]: This is storybook config, so it's no problem :)

File: build/css.js (hunk 3 above)
Comment (WANZARGEN) [reply build/css.js:6]: `postcss.config.cjs` file in root is common js module. it is different from `.storybook/postcss.config.js`

File: package.json (hunk 4 above)
Comment (WANZARGEN) [reply package.json]: Oh thanx!

**Review by sulmoJ — APPROVED** [review]
Thank you for your hard work!
I have a question.
The `build:watch` command looks the same, is the link method the same as before?

**Pull request merged**